### ISSUE TITLE
outputs设置多个loss，最终log中输出的loss是什么

### ISSUE THREAD

**Issue by colin1988** [PaddlePaddle/Paddle#2913]
模型配置中output输出多个loss，代码如下：
```python
seq_loss = cross_entropy(input = seq_label, label = target)
crf_l = crf_layer(
       name='crf',
         size=label_dict_len,
         input=feat,
         label=target,
         param_attr=ParameterAttribute(
             name='crfw', learning_rate=mix_hidden_lr))
```
**outputs(seq_loss, crf_l)**

训练日志中输出单个loss
Batch=3990 samples=598500 **AvgCost**=1.87198 **CurrentCost**=1.91031 Eval: __sum_evaluator_0__=0.0453391  CurrentEval: __sum_evaluator_0__=0.0440848

请问log输出的单个loss是如何通过outputs中设置的多个loss计算出来的，优化又是如何进行的？

**Comment by colin1988** [comment]
另外，对sequence数据使用cross_entropy loss，log输出的loss是如何通过单个时间步的loss归约而来，求和还是平均，目前猜测是求和，希望得到确认。

**Comment by lcy-seso** [comment]
1. 当指定多个损失函数时，日志中的输出是把多个损失函数求和。https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/trainer/TrainerInternal.cpp#L137
    ```python
    cost = Argument::sum(*outArgs);
    ```
    上面代码中 `outArgs` 就是配置中指定的一个或者多个损失函数，Paddle 目前不会分别输出每个损失函数的值。

2. 多个损失函数的优化依然遵循BP 。
3. 对sequence数据使用cross_entropy loss，log 输出的 loss 对所有时间步求和。

**Comment by Canpio** [comment]
问题已经解答，因此关闭issue

**Comment by chenbblei** [comment]
楼主好，请问您怎么实现多个loss结合在一起做训练的，比如loss1，loss2,我想按照loss = loss1*0.5+loss2*.05的方式做loss，如何实现
